### PR TITLE
[JUJU-1175] Silence issue with teardown in kubeflow integration test

### PR DESCRIPTION
**Pull request by benhoyt**
There was an error in teardown due to a bug in the
metacontroller-operator charm's "remove" hook:
https://github.com/canonical/metacontroller-operator/issues/5

For now, just avoid doing a proper teardown (testing that charm
"remove" hooks work is not what we're trying to test here) and let the
Microk8s teardown clean up.

I've also added a "juju status" in the failure case to show what
applications it was stuck on. (In this case I was able to repro
locally to find that it was the metacontroller-operator charm.)

Jira link: https://warthogs.atlassian.net/browse/JUJU-1175

### CODE REVIEW

File: acceptancetests/assess_caas_deploy_kubeflow.py [hunk 1]
```diff
@@ -410,8 +411,12 @@ def success_hook():
             caas_client.kubectl('get', 'sa,roles,clusterroles,rolebindings,clusterrolebindings', '-oyaml', '-A'),
         )
 
+    def success_hook():
+        dump_success_logs()
+        caas_client.ensure_cleanup()
```
Comment (ycliuhw) [inline acceptancetests/assess_caas_deploy_kubeflow.py:416]: Could we have a comment here to explain why we need to delete the namespace before we have proper teardown (probably just copy the comment below for `bs_manager.keep_env = False`)?

**Review by ycliuhw — APPROVED** [review]
LGTM ty

File: acceptancetests/assess_caas_deploy_kubeflow.py (hunk 1 above)
Comment (benhoyt) [reply acceptancetests/assess_caas_deploy_kubeflow.py]: Done.

**Review by wallyworld — APPROVED** [review]

**Pull request merged**